### DISCUSSION TITLE
Upgrading grapesjs breaks Typescript: How do I use Type Definitions

### DISCUSSION THREAD

**Discussion by stljeff1** [GrapesJS/grapesjs#4533] [Q&A]
I am having all kinds of issues (like [this](https://github.com/artf/grapesjs/discussions/4507)) with blocks, and have decided to upgrade.

I am using Yarn, Angular, and Typescript in my app. After upgrading grapes using yarn (`yarn upgrade grapesjs --latest`), I get the following error:

```
 error TS2339: Property 'init' does not exist on type 'typeof import("/home/projects/myproject/node_modules/grapesjs/index")'.
    
    369   const grapesEditor = grapesjs.init({

```

Here is my import statement:

`import * as grapesjs from 'grapesjs';`

I am a novice with Typescript, so I feel trapped and confused by this bug. Do I have to import custom type definitions or something? 

Any insight it much appreciated.

**Comment by stljeff1** [comment]
here is my current `tsconfig.ts` file

```
{
  ....
   "typeRoots": [
      "./node_modules/@types",
      "./node_modules/grapesjs"
    ]
}

```

adding grapes to typeRoots causes VS code to wig out. it trows te following   error:

`Cannot find type definition file for 'dist'. The file is in the program because: Entry point for implicit type library 'dist'ts`

researching this error suggests that this is a bug of some kind, and restarting vs code or deteling `node_modules` will fix  the error, but that is not my experience

**Comment by artf** [comment]
@stljeff1 Try to simply update your import
```diff
-import * as grapesjs from 'grapesjs';
+import grapesjs from 'grapesjs';
```
By using the latest grapesjs version I don't see the issue you're mentioning.
![ts-init](https://user-images.githubusercontent.com/11614725/188411530-9c2b619c-6f7d-4fae-827f-49bfec3988b7.png)

**Reply by IRediTOTO** [reply]
How about this? I am trying use `gjs-blocks-basic`
![image](https://user-images.githubusercontent.com/15695252/189000819-05a16605-ca1f-400a-886d-5df2e72012c5.png)

**Comment by stljeff1** [comment]
I went with @artf's suggested update. It seems to be working again. 

I believe I  was experiencing some sort of caching issue. I saw posts on stackoverflow about VS Code not picking up newly added type definitions. I upgraded from 16.2-ish, so the type definitions were new. I have no idea, all i know is I tried both

 `import grapesjs from 'grapesjs';` as per the official documentation

and the original

`import * as grapesjs from 'grapesjs';`

with both creating strange errors that didn't make sense. I'm working within a much larger application, so I figured it was beyond my understanding. I let some time go by until i tried again yesterday when VS code actually recognized the init function

**Comment by stljeff1** [comment]
@artf - it does seem the Type definitions do not match up with whats in the source code.

For example, the `EditorConfig` definition is inaccurate. 

1. its missing `page` and `pageManager` properties.
2. the `log` propery type is wrong. It currently does not allow for a boolean value, which is a valid value as per [documentation](https://grapesjs.com/docs/api/pages.html#pages)

This is just my initial findings. I know you are not a typescript developer, so I don't consider this a bug. However, i am excited to use the new Type definitions, and look forward to using them. I will try to dig deeper to find more inconsistencies and create a PR to help fix... for now, i just wanted to comment on the few inconsistencies i've seen  in case it can be verified by others.

**Reply by artf** [reply]
Hi @stljeff1 the TS thing it's still a work in progress. We're slowly migrating JS files to TS and maintaining some common types in `index.d.ts` to have at least something typed before the migration is completed. If you're willing to help with the TS migration that would be great.
